CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -129,6 +129,16 @@ public class Child: IdentifierConvertible {
   /// The first line of the child's documentation
   public let documentationAbstract: String
 
+  /// The documentation note to print for an experimental feature.
+  public var experimentalDocNote: SwiftSyntax.Trivia {
+    let comment = experimentalFeature.map {
+      """
+      - Experiment: Requires experimental feature `\($0.token)`.
```

Comment:
IIRC the `- Experiment:` comment in docc is intended for an experiment a user of the API might perform to get familiar with it. That’s  not what experimental means here. I would suggest using `- Note:` or `- Warning:` instead.